session/ephemeral.go

```diff
@@ -0,0 +1,57 @@
+package session
+
+import (
+	"errors"
+
+	"github.com/Shopify/gozk"
+)
+
+// CreateAndMaintainEphemeral creates an ephemeral znode with the given
+// path+data, and signals the provided channel when it has stopped. This
+// indicates the node no longer exists, either because the connection was
+// closed or an error occurred. As long as the channel has not been signalled,
+// the caller can reasonable expect that the ephemeral node still exists.
+//
+// This is not an appropriate construct to use for locking, as a partition will
+// not be immediately reported to the caller; the code will wait for a
+// reconnect or expiry before notifying.
+func (z *ZKSession) CreateAndMaintainEphemeral(path, data string, dead chan<- error) error {
+	doCreate := func() error {
+		_, err := z.conn.Create(path, data, zookeeper.EPHEMERAL, defaultACLs)
+		return err
+	}
+
+	if err := doCreate(); err != nil {
+		return err
+	}
+
+	evs := make(chan ZKSessionEvent)
+	z.Subscribe(evs)
+
+	go func() { dead <- maintainEphemeral(evs, doCreate) }()
```

Comment:
Do you also want to make this cancellable? Perhaps with an abort channel. I know this will complicate the function signature but as it currently stands there is no way for the caller to abort maintaining the ephemeral node.